vizro-core/docs/pages/user_guides/container.md

```diff
@@ -105,3 +126,173 @@ To add a [`Container`][vizro.models.Container] to your page, do the following:
+
+!!! note
+
+    Note that an almost identical layout can also be achieved using [`Page.layout`](layouts.md), see the
+    [advanced grid example](layouts.md#grid-advanced-example) on how this can be done. Here we use `Container` instead
+    of `Page.layout` because we need finer control of selected grid areas (e.g. adding different titles to each subgrid
+    and have different spacing between groups of components).
+
+!!! example "Nested Containers"
+    === "app.py"
+        ```py
+
+        import vizro.models as vm
+        import vizro.plotly.express as px
+        from vizro import Vizro
+
+        df = px.data.gapminder()
+
+        nested_container_one = vm.Container(
+            title="Nested Container I",
+            layout=vm.Layout(grid=[[0, 1], [2, 2]]),
+            components=[
+                vm.Graph(
+                    figure=px.line(
+                        df,
+                        title="Graph 1 - Nested Container I",
+                        x="year",
+                        y="lifeExp",
+                        color="continent",
+                    ),
+                ),
+                vm.Graph(
+                    figure=px.scatter(
+                        df,
+                        title="Graph 2 - Nested Container I",
+                        x="gdpPercap",
+                        y="lifeExp",
+                        size="pop",
+                        color="continent",
+                    ),
+                ),
+                vm.Graph(
+                    figure=px.box(
+                        df,
+                        title="Graph 3 - Nested Container I",
+                        x="continent",
+                        y="lifeExp",
+                        color="continent",
+                    ),
+                ),
+            ],
+        )
+
+        nested_container_two = vm.Container(
+            title="Nested Container II",
+            layout=vm.Layout(grid=[[0, 1]]),
+            components=[
+                vm.Graph(
+                    figure=px.line(
+                        df,
+                        title="Graph 4 - Nested Container II",
+                        x="year",
+                        y="lifeExp",
+                        color="continent",
+                    ),
+                ),
+                vm.Graph(
+                    figure=px.scatter(
+                        df,
+                        title="Graph 5 - Nested Container II",
+                        x="gdpPercap",
+                        y="lifeExp",
+                        size="pop",
+                        color="continent",
+                    ),
+                ),
+            ],
+        )
+
+        page = vm.Page(
+            title="Nested Containers",
+            components=[
+                vm.Container(
+                    title="Container Title",
+                    layout=vm.Layout(grid=[[0, 1]], col_gap="80px"),
```

Comment:
It's not even clear up to this point that these containers are nested. Maybe it makes sense to have a code pop-out comment here that shows that one could very easily (just 2 brackts or deleting the grid) out the two groups of charts below one another.

If we do, would maybe refer to that also in the above suggestion I made

Reply:
Hmm.. could you clarify? I am not sure if I understand above. I've renamed the variables to `child_container_one` and `child_container_two`, which I think makes it quite clear that the components of the Container itself are Containers now. Does that help?

Reply:
@antonymilne  - just had a chat with @maxschulz-COL and I get his confusion of why this example is used for nesting as theoretically we could also achieve the same layout by adding the two child container directly into `Page.components`.

Not sure what the right approach here is to make it clear that nesting is possible, but might not always be necessary? Should we actually just kill this section of nested containers and reuse the example here with two containers inside `Page.components`?

in general the nested containers make more sense when the Tabs come in. Before that one could probably achieve most of the layouts with top-level containers and page.layout only?

Reply:
See comment above - my vote for the resolution of this

Reply:
I definitely also see the cause of confusion here. The problem is that the use of these nested containers is indeed quite limited until we add `controls` to the `Container` model. At the moment, in terms of the overall grid created, there's nothing you can do with nesting `Container` models that you can't already do with `Page.layout`.

I would still keep this example because the `col_gap` here does enable something you can't do just purely through `Page.layout`. And in future when we add `controls` the example will become much more interesting and clearer what the point of containers is.

Reply:
Hmm yes I definitely also see the cause of confusion here and I also saw had the same thoughts as @maxschulz-COL when I read through it again. The problem is that the use of these nested containers is indeed quite limited until we add `controls` to the `Container` model. At the moment, in terms of the overall grid created, there's nothing you can do with nesting `Container` models that you can't already do with `Page.layout`.

Let me comment my suggestion on the other conversation to keep it all on one place.